src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/mongo/AgoraMongoDao.scala

```diff
@@ -64,10 +62,22 @@ class AgoraMongoDao(collection: MongoCollection) extends AgoraDao {
   }
 
   def find(query: Imports.DBObject) = {
-    (for (entity <- collection.find(query)) yield grater[AgoraEntity].asObject(entity)).toVector
+    (for (dbObject <- collection.find(query)) yield MongoDbObjectToEntity(dbObject)).toVector
+  }
+
+  override def find(entity: AgoraEntity): Seq[AgoraEntity] = find(EntityToMongoDbObject(entity))
+
+  override def findSingle(entity: AgoraEntity): AgoraEntity = {
+    val entityVector = find(EntityToMongoDbObject(entity))
+    entityVector.length match {
+      case 1 => entityVector.head
+      case 0 => throw new Exception("Found 0 documents matching: " + entity.toString)
+      case x if x > 1 => throw new Exception("Found >1 documents matching: " + entity.toString)
```

Comment:
What if you get -1? Probably that can't happen and we can just use _.

But what if mongo uses negatives as an error code or something? In that case, this match is not exhaustive